mypy/server/astdiff.py

```diff
@@ -234,16 +234,16 @@ def snapshot_definition(node: SymbolNode | None, common: SymbolSnapshot) -> Symb
     The representation is nested tuples and dicts. Only externally
     visible attributes are included.
     """
-    if isinstance(node, FuncBase):
+    if isinstance(node, SYMBOL_FUNCBASE_TYPES):
         # TODO: info
         if node.type:
-            signature = snapshot_type(node.type)
+            signature: tuple[object, ...] = snapshot_type(node.type)
         else:
             signature = snapshot_untyped_signature(node)
         impl: FuncDef | None = None
         if isinstance(node, FuncDef):
             impl = node
-        elif isinstance(node, OverloadedFuncDef) and node.impl:
+        elif node.impl:
```

Comment:
I'm cautious about this change.

Reply:
Why? :)

Reply:
Actually never mind as long as it type checks. I was worried about if we revert above isinstance (if we fix the trait reachability thing) and then this might not be guaranteed... but in that case mypy will tell us the attribute doesn't exist...